### ISSUE TITLE
Proper Fahrenheit support

### ISSUE THREAD

**Issue by nfarina** [nfarina/homebridge#71]
(Moving discussion from #70, #65)

So it looks like HomeKit thermostats can be set to Celsius or Fahrenheit directly. Notice how in the HomeKit Accessory Simulator you can set the unit type. And in the JSON service info, there's a TEMPERATURE_UNITS_CTYPE setting, as well as a unit: property for the TARGET_TEMPERATURE_CTYPE.

Does anyone know how this is supposed to work? Is user supposed to say something like "Change my thermostat units to fahrenheit"? Or is this an innate setting of the accessory itself, and if so what is the correct value?

**Comment by Danimal4326** [comment]
I tried to force the units to F in the Nest.js but no luck.  It still thought it was in Celsius.
When changing temperature via Siri, it still converts to C before setting it.  When I say "set the home temperature to 76" she responds "setting temperature to 75.2" (or something like that)

**Comment by nfarina** [comment]
Does the same thing happen when using the HomeKit Accessory Simulator on OS X and setting it to Fahrenheit?

**Comment by Danimal4326** [comment]
I have not tried the simulator.

**Comment by fordracerguy** [comment]
I just loaded the simulator. It has a setting for C or F. Set it to F. This is much more accurate. In my 85F example, Siri said "OK Chris, I set the thermostat to 84.9F.". Remember in my previous posting, it went to 84.2. 71F goes to 71.1F, All other temperatures tested, 75, 86, etc, seem to work perfectly (no decimal point). Unfortunately I can't see a log of what is actually being sent from the phone, I can only use what Siri is reporting. It seems better, but still has issues... I think there still may be an issue with HAP in the UNIT type sent to HomeKit when pairing.
